llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -326,6 +326,18 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
     switch (Kind) {
     default:
       break;
+    case TTI::SK_InsertSubvector:
+      if (auto *FSubTy = dyn_cast<FixedVectorType>(SubTp)) {
```

Comment:
Sorry just noticing this now, I think this should be a cast since it shouldn't be possible to insert a scalable `SubTp` into `Tp`
```suggestion
      auto *FSubTy = cast<FixedVectorType>(SubTp);
```

Reply:
Ok, will co, tried to add an extra level of protection here